private/buf/bufformat/formatter.go

```diff
@@ -1261,7 +1296,9 @@ func (f *formatter) writeOpenBracePrefix(openBrace ast.Node) {
 	info := f.fileNode.NodeInfo(openBrace)
 	if info.LeadingComments().Len() > 0 {
 		f.writeInlineComments(info.LeadingComments())
-		f.Space()
+		if info.LeadingWhitespace() != "" {
```

Comment:
This is part of the logic that tries to preserve whitespace conventions. You'll see this condition sprinkled in a few places.